pennylane/optimize/qng.py

```diff
@@ -155,19 +156,26 @@ def __init__(self, stepsize=0.01, diag_approx=False, lam=0):
         self.metric_tensor = None
         self.lam = lam
 
-    def step_and_cost(self, qnode, x, recompute_tensor=True, metric_tensor_fn=None):
+    def step_and_cost(
+        self, qnode, *args, grad_fn=None, recompute_tensor=True, metric_tensor_fn=None, **kwargs
+    ):
         """Update the parameter array :math:`x` with one step of the optimizer and return the
         corresponding objective function value prior to the step.
 
         Args:
             qnode (QNode): the QNode for optimization
-            x (array): NumPy array containing the current values of the variables to be updated
+            *args : variable length argument list for qnode
+            grad_fn (function): optional gradient function of the
+                qnode with respect to the variables ``*args``.
+                If ``None``, the gradient function is computed automatically.
+                Must return the same shape of tuple [array] as the autograd derivative.
```

Comment:
I realise this is the same line as the other optimizer methods have, but to be honest, it isn't very clear to me - the `tuple[array]` placement in the sentence I find a bit confusing.

Maybe the following?

> Must return ``tuple[array]`` with the same number of elements as ``*args``. Each array of the tuple
> should correspond to the shape of the corresponding argument.

Reply:
Will replace this in other optimizers as well, I agree it wasn't so clear..